sdk/typescript/runtime/main.go

```diff
@@ -184,29 +179,57 @@ func (t *TypescriptSdk) CodegenBase(ctx context.Context, modSource *dagger.Modul
 	base = base.
 		// Add template directory
 		WithMountedDirectory("/opt/module", dag.CurrentModule().Source().Directory(".")).
-		// Mount users' module with SDK sources and generated client in it.
-		WithMountedDirectory(ModSourceDirPath,
+		// Mount user's module configuration (without sources) and the generated client in it.
+		WithDirectory(ModSourceDirPath,
 			dag.Directory().WithDirectory("/", modSource.ContextDirectory(), dagger.DirectoryWithDirectoryOpts{
 				Include: []string{
-					fmt.Sprintf("%s/**", t.moduleConfig.subPath),
+					fmt.Sprintf("%s/package.json", t.moduleConfig.subPath),
+					fmt.Sprintf("%s/*lock*", t.moduleConfig.subPath),
+					fmt.Sprintf("%s/tsconfig.json", t.moduleConfig.subPath),
+					fmt.Sprintf("%s/pnpm-workspace.yaml", t.moduleConfig.subPath),
+					fmt.Sprintf("%s/.yarnrc.yml", t.moduleConfig.subPath),
 				},
-			}).
-				WithDirectory(filepath.Join(t.moduleConfig.subPath, GenDir), sdk),
-		).
+			})).
+		WithDirectory(filepath.Join(t.moduleConfig.modulePath(), GenDir), sdk).
 		WithWorkdir(t.moduleConfig.modulePath())
 
-	ctr, err := t.setupModule(ctx, base)
-	if err != nil {
-		return nil, fmt.Errorf("failed to setup module: %w", err)
-	}
+	base = t.configureModule(base)
 
 	// Generate the appropriate lock file
-	ctr, err = t.generateLockFile(ctr)
+	base, err = t.generateLockFile(base)
 	if err != nil {
 		return nil, fmt.Errorf("failed to generate lock file: %w", err)
 	}
 
-	return ctr, nil
+	// Install dependencies if needed.
+	if installDep {
+		base, err = t.installDependencies(base)
+		if err != nil {
+			return nil, fmt.Errorf("failed to install dependencies: %w", err)
+		}
+	}
+
+	// Add user's source files
+	base = base.WithDirectory(ModSourceDirPath,
+		dag.Directory().WithDirectory("/", modSource.ContextDirectory(), dagger.DirectoryWithDirectoryOpts{
+			// Include the rest of the user's module except config files to not override previous steps & SDKs.
+			Exclude: []string{
+				fmt.Sprintf("%s/package.json", t.moduleConfig.subPath),
+				fmt.Sprintf("%s/*lock*", t.moduleConfig.subPath),
```

Comment:
same comment.

Also, is this `exclude` really needed? Those files are very small and should 100% match still at this point right?

Reply:
Yes, to not overwritten potential update on package.json and lockfiles

Reply:
Done in [52fd637](https://github.com/dagger/dagger/pull/8236/commits/52fd6376526ba1e9fdec6e1d0c307a48cf370754)